src/process/io/locks/PathLockHandler.cpp

```diff
@@ -0,0 +1,82 @@
+/**
+ * Copyright (c) 2019 DKFZ - ODCF
+ *
+ * Distributed under the MIT License (license terms are at https://github.com/dkfz-odcf/FastqIndEx/blob/master/LICENSE.txt).
+ */
+#include "PathLockHandler.h"
+#include <mutex>
+#include <sys/file.h>
+#include <utility>
+using std::lock_guard;
+using std::mutex;
+
+PathLockHandler::PathLockHandler(path file)  {
```

Comment:
Could this be a `FileLockHandler`? This uses `flock` which is based on some filesystem internal mechanism (as compared to e.g. procmails `lockfile` tool that simply uses a dedicated file with a specific path). This make this implementation pretty "file"ish rather than "path"ish, or not?

Furthermore, is the mutex stuff in here worth mentioning in the class name?